src/Tasks/GenerateResource.cs

```diff
@@ -995,7 +995,8 @@ private bool TransformResourceFilesUsingResGen(List<ITaskItem> inputsToProcess,
 
             // We need to do a whole lot of work to make sure that we're not overrunning the command line ... UNLESS
             // we're running ResGen 4.0 or later, which supports response files. 
-            if (!_resgenPath.Equals(Path.GetDirectoryName(NativeMethodsShared.GetLongFilePath(ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("resgen.exe", TargetDotNetFrameworkVersion.Version35))), StringComparison.OrdinalIgnoreCase))
+            if (NativeMethodsShared.IsWindows &&
```

Comment:
Just for the record, the Mono ResGen doesn't support response files on any platform. Neither does it support the /r:<reference> syntax. It is basically unusable by msbuild at this moment. We use a customized and extended version for our purposes. The response file requirement is something from ResGen.cs, where it is disabled for non-Windows system. This patch just aligns the "if" conditions there with the ones in GenerateResource.cs. I would be equally fine with removing the conditions in ResGen.cs and enabling the response files on all platforms, but I suspected someone put it there for a reason beyond my understanding.